types/rest/rest.go

```diff
@@ -0,0 +1,284 @@
+// Package rest provides HTTP types and primitives for REST
+// requests validation and responses handling.
+package rest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types"
+)
+
+// GasEstimateResponse defines a response definition for tx gas estimation.
+type GasEstimateResponse struct {
+	GasEstimate uint64 `json:"gas_estimate"`
+}
+
+// UnjailReq request unjailing
+type UnjailReq struct {
```

Comment:
Why is this here?

Reply:
I moved all requests into this package

Reply:
imho I don't think we should be doing this -- but I need to think on this some more after some coffee.

Reply:
Agree with @alexanderbez here. I think each `client/rest` package should also have a `types` that exports the structs to call the endpoints.